monitor_state_dict.py

```diff
@@ -0,0 +1,190 @@
+
+    # Game related
+    GameLanguage:           str                             # From `Fileheader`
+    GameVersion:            str                             # From `Fileheader`
+    GameBuild:              str                             # From `Fileheader`
+    Horizons:               bool                            # Does the player have Horizons?
+    Odyssey:                bool                            # Have we detected Odyssey?
+
+    # Multi-crew
+
+    Captain:                str                   # If on a crew, the captian's name
+    Role:                   str                   # Role in crew
+
+    # Cmdr state
+    FID:                    str                             # Frontier CMDR ID
+    Friends:                Set[str]                        # Online Friends
+    Credits:                int
+    Loan:                   int
+
+    # (A_D) One day I will change this to be a NamedTuple. But for now it will suffice to state that:
+    # (Rank, RankProgress) | Literal['Known', 'Invited' (or any other of the possible states that ISNT a rank number)]
+    Engineers:              Dict[str, Union[str, Tuple[int, int]]]
+    Rank:                   Dict[str, Tuple[int, int]]      # (RankMajor, RankProgress)
+    Reputation:             Dict[str, float]                # Superpower -> level
+    Statistics:             Dict[Any, Any]                  # This is very freeform.
+
+    # Engineering Materials
+    Raw:                    DefaultDict[str, int]
+    Encoded:                DefaultDict[str, int]
+    Manufactured:           DefaultDict[str, int]
+
+    # Ship
+    ShipID:                 int
+    ShipIdent:              str
+    ShipName:               str
+    ShipType:               str
+
+    HullValue:              int
+    ModulesValue:           int
+    Rebuy:                  int
+    Modules:                Dict[str, ModuleDict]
+    ModuleInfo:             MutableMapping[Any, Any]        # From the game, freeform
+
+    # Cargo (yes technically its on the cmdr not the ship but this makes more sense.)
+    CargoJSON:              MutableMapping[str, Any]        # Raw data from the last cargo.json read
+    Cargo:                  DefaultDict[str, int]
+
+    # Navigation
+    NavRoute:               NavRouteDict                    # Last route plotted
+    Body:                   str
+    BodyType:               str
+    Taxi:                   bool
+    Dropship:               bool
+
+    # Odyssey
+    OnFoot:                 bool
+    Component:              DefaultDict[str, int]
+    Item:                   DefaultDict[str, int]
+    Consumable:             DefaultDict[str, int]
+    Data:                   DefaultDict[str, int]
+    BackPack:               OdysseyBackpack
+    BackpackJSON:           MutableMapping[str, Any]        # Direct from Game
+    ShipLockerJSON:         MutableMapping[str, Any]        # Direct from Game
+
+    SuitCurrent:            Optional[SuitDict]
```

Comment:
Without checking the code that uses this, I am wondering if we can't default this to `{}` and lose the `Optional`.  Same for SuitLoadoutCurrent below.

Reply:
specifically because the "default" cannot be `{}`, it must be a dict that matches `SuitDict`, so the minimum default is:
```python
{
'name': '',
'locName': '',
'weaponrackId': 0,
'id': None,
'suitId': 0,
'mods': [],
}
```
which is not going to be falsy